### ISSUE TITLE
Use 're2' package to prevent ReDoS?

### ISSUE THREAD

**Issue by phormio** [ajv-validator/ajv-formats#7]
According to the [Ajv documentation](https://github.com/ajv-validator/ajv/blob/bd8d86579386cb58854e9db74a5dc7b2d1894806/docs/security.md#redos-attack):

> some formats that ajv-formats package implements use regular expressions that can be vulnerable to ReDoS attack

Having you considered using the [`re2`](https://www.npmjs.com/package/re2) package to eliminate this risk?  According to its [documentation](https://github.com/google/re2/wiki/WhyRE2):

> RE2 was designed and implemented with an explicit goal of being able to handle regular expressions from untrusted users without risk. One of its primary guarantees is that the match time is linear in the length of the input string.

**Comment by shumkov** [comment]
Yeah, we are thinking about the same to secure Ajv. It would be cool to have such an option.

**Comment by epoberezkin** [comment]
This option probably belongs in ajv - this package both uses formats in some functions and also provides regular expressions that ajv uses directly, so it would have to somehow use different branches depending on ajv option...

**Comment by shumkov** [comment]
Make sense. Actually, RE2 npm package is almost a drop-in replacement for `RegExp` constructor. We are planning to monkey-patch (something like `global.RegExp = RE2`) ajv on the first iteration.
